src/core/ddsi/src/q_entity.c

```diff
@@ -3894,6 +4209,132 @@ static void add_proxy_builtin_endpoints(
 #undef LTE
 }
 
+#ifdef DDSI_INCLUDE_SECURITY
+
+void handshake_end_cb
+(
+  struct ddsi_handshake *handshake,
+  const struct ddsi_guid *lpguid,
+  const struct ddsi_guid *ppguid,
+  nn_wctime_t timestamp,
+  enum ddsi_handshake_state result)
+{
+  struct proxy_participant *proxypp;
+  struct participant *pp;
+  int64_t shared_secret;
+  int64_t permissions_hdl;
+  struct q_globals *gv;
+
+  DDSRT_UNUSED_ARG(timestamp);
+
+  assert(handshake);
+  assert(lpguid);
+  assert(ppguid);
+
+  gv = ddsi_handshake_get_globals(handshake);
```

Comment:
Based on the code in this PR it would make more sense for ``gv`` to be passed in, rather than extracted from the ``handshake`` object.